plugins/optimization-detective/readme.txt

```diff
@@ -37,16 +46,47 @@ When the `WP_DEBUG` constant is enabled, additional logging for Optimization Det
 
 Fires when the Optimization Detective is initializing. This action is useful for loading extension code that depends on Optimization Detective to be running. The version of the plugin is passed as the sole argument so that if the required version is not present, the callback can short circuit.
 
-**Filter:** `od_breakpoint_max_widths` (default: [480, 600, 782])
+**Action:** `od_register_tag_visitors` (argument: `OD_Tag_Visitor_Registry`)
+
+Fires to register tag visitors before walking over the document to perform optimizations.
 
-Filters the breakpoint max widths to group URL metrics for various viewports. Each number represents the maximum width (inclusive) for a given breakpoint. So if there is one number, 480, then this means there will be two viewport groupings, one for 0<=480, and another >480. If instead there were three provided breakpoints (320, 480, 576) then this means there will be four groups:
+For example, to register a new tag visitor that targets `H1` elements:
+
+`
+<?php
+add_action(
+	'od_register_tag_visitors',
+	static function ( OD_Tag_Visitor_Registry $registry ) {
+		$registry->register(
+			'my-plugin/h1',
+			static function ( OD_Tag_Visitor_Context $context ): bool {
+				if ( $context->processor->get_tag() !== 'H1' ) {
+					return false;
+				}
+				// Now optimize based on stored URL Metrics in $context->url_metric_group_collection.
+				// ...
+
+				// Returning true causes the tag to be tracked in URL Metrics. If there is no need
+				// for this, as in there is no reference to $context->url_metric_group_collection
+				// in a tag visitor, then this can instead return false.
+				return true;
+			}
+		);
+	}
+);
+`
 
- 1. 0-320 (small smartphone)
- 2. 321-480 (normal smartphone)
- 3. 481-576 (phablets)
- 4. >576 (desktop)
+Refer to [Image Prioritizer](https://github.com/WordPress/performance/tree/trunk/plugins/image-prioritizer) and [Embed Optimizer](https://github.com/WordPress/performance/tree/trunk/plugins/embed-optimizer) for real world examples of how tag visitors are used. Registered tag visitors need only be callables, so in addition to providing a closure you may provide a `callable-string` or even a class which has an `__invoke()` method.
 
-The default breakpoints are reused from Gutenberg which appear to be used the most in media queries that affect frontend styles.
+**Filter:** `od_breakpoint_max_widths` (default: `array(480, 600, 782)`)
+
+Filters the breakpoint max widths to group URL Metrics for various viewports. Each number represents the maximum width (inclusive) for a given breakpoint. So if there is one number, 480, then this means there will be two viewport groupings, one for 0\<=480, and another \>480. If instead there are the two breakpoints defined, 480 and 782, then this means there will be three viewport groups of URL Metrics:
+
+1. Mobile: 0-480px
+2. Tablet: 481-782px
+3. Desktop: \>782px
```

Comment:
I think you should avoid using a list for this and put it inline in the text, just like you did for the other example of two widths.

The list makes it seem somehow "official". For instance, the example doesn't mention giving the breakpoints a name, but the list has names, which I think makes it even more confusing. So I think it would be best to only refer to:
```suggestion
Filters the breakpoint max widths to group URL Metrics for various viewports. Each number represents the maximum width (inclusive) for a given breakpoint. So if there is one number, 480, then this means there will be two viewport groupings, one for 0\<=480, and another \>480. If instead there are the two breakpoints defined, 480 and 782, then this means there will be three viewport groups of URL metrics, one for 0\<=480, another 481\<=782, and another \>782.
```

Reply:
I made a small addition on top of this suggestion: e9827afc